src/Dfe.PlanTech.Web/Middleware/ServiceExceptionHandlerMiddleWare.cs

```diff
@@ -22,10 +22,10 @@ static string GetRedirectUrlForException(Exception? exception) =>
         exception switch
         {
             null => UrlConstants.Error,
-            ContentfulDataUnavailableException => UrlConstants.ServiceUnavailable,
+            ContentfulDataUnavailableException => UrlConstants.NotFound,
```

Comment:
Does this change only affect pages? If not, do we want this to be this? What are the implications?

Reply:
Good point. I've created a separate exception for Contentful pages [a2b0b45](https://github.com/DFE-Digital/sts-plan-technology-for-your-school/pull/838/commits/a2b0b45a6eee2e2383baefcfbe97a80676ff4960)

Reply:
What's the use case for `PageNotFoundException` vs `ContentfulPageUnavailableException`?

Reply:
You're right, its just more confusing if anything. I'll just throw a page not found exception

Reply:
Better IMO. But here's a question; why are we even throwing an exception if the page isn't found? The method already returns a nullable page, but actually with its current implementation it'll never return a null page.

In addition, currently it'll throw _2_ exceptions if a page isn't found. Exceptions are kinda expensive. Not expensive enough for this to actually be an issue, but could we do anything to improve the process here?

Reply:
Tried returning a null page and just handling that in the controller. How does that approach look?

Reply:
Much better mate, just couple more things:

1. I'd still throw the `ContentfulDataUnavailableException` in the try catch. If there's an exception being thrown, then it's unlikely to be a 404 error, it's going to be a legitimate (hopefully transient) problem
2. Might be good to log the missing page? Just as an `information` level - just incase we need to debug later
3. Do we still need `PageNotFoundException`?

Reply:
Jolly good 👍